src/4-episode-taskeither/index.tsx

```diff
@@ -46,7 +45,24 @@ export function useTaskEither<E, A>() {
     task().then((either) => pipe(either, O.some, setValue))
   }, [])
 
-  return [value, isLoading, run] as const
+  const match = React.useCallback(
+    <B, C, D, F>(
+      onNone: () => B,
+      onLoading: () => C,
+      onError: (error: E) => D,
+      onSuccess: (value: A) => F,
+    ) =>
+      pipe(
+        value,
+        O.matchW(
```

Comment:
💭  what is `matchW` doing here, I assume widening the possibilities within the error channel yeah?

Reply:
It's actually widening the return value in this case